frontend/src/state/reducers/userReducer.ts

```diff
@@ -18,34 +18,14 @@ import {
 } from "../actions/userActions";
 import { IUserData } from "../../models/types";
 import { ScheduleCourse } from "../../../../common/types";
-import {
-  convertToDNDCourses,
-  sumCreditsFromList,
-  numToTerm,
-  getNextTerm,
-  produceWarnings,
-} from "../../utils";
 
 export interface UserState {
-  user: IUserData;
+  user?: IUserData;
   completedRequirements: IRequiredCourse[]; // only used in onboarding flow
```

Comment:
This pattern better reflects how we're using this reducer in conjunction with our api